src/app/application.cpp

```diff
@@ -139,11 +139,18 @@ void Application::torrentFinished(BitTorrent::TorrentHandle *const torrent)
 
     // AutoRun program
     if (pref->isAutoRunEnabled()) {
-        QString program = pref->getAutoRunProgram().trimmed();
-        // Replace %f by torrent path
-        program.replace("%f", torrent->savePathParsed());
-        // Replace %n by torrent name
-        program.replace("%n", torrent->name());
+        QString program = pref->getAutoRunProgram();
+        int file_count = torrent->filesCount();
+
+        program.replace("%N", torrent->name());
+        if (file_count == 1)
+            program.replace("%F", torrent->fileName(0));
```

Comment:
I don't know what's the behavior in uTorrent when `file_count > 1` but I think we should replace `%F` with an empty string.